scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,263 @@
+package org.oppia.android.scripts
+
+import java.io.File
+import java.io.FileInputStream
+import org.oppia.android.scripts.proto.FilenameChecks
+import org.oppia.android.scripts.proto.FilenameCheck
+import org.oppia.android.scripts.proto.FileContentChecks
+import org.oppia.android.scripts.proto.FileContentCheck
+import org.oppia.android.scripts.ScriptResultConstants
+import com.google.protobuf.MessageLite
+
+/** Script for ensuring that prohibited file contents and
+ *  file naming patterns are not present in the codebase.
+ */
+class RegexPatternValidationCheck {
+  companion object {
+    @JvmStatic
+    fun main(vararg args: String) {
+      /** path of the repo to be analyzed. */
+      val repoPath = args[0] + "/"
+
+      /** a list of all allowed directories in the repo to be analyzed. */
+      val allowedDirectories = args.drop(1)
+
+      /** a list of all files in the repo to be analyzed. */
+      val searchFiles = RepoFile.collectSearchFiles(
+        repoPath = repoPath,
+        allowedDirectories = allowedDirectories
+      )
+
+      /** check if the repo has any filename failure. */
+      val hasFilenameCheckFailure = retrieveFilenameChecks()
+        .fold(initial = false) { isFailing, check ->
+          val checkFailed = checkProhibitedFileNamePattern(
+            repoPath = repoPath,
+            searchFiles = searchFiles,
+            prohibitedFilenameObj = check,
+          )
+          return@fold isFailing || checkFailed
+        }
+
+      /** check if the repo has any file content failure. */
+      val hasFileContentCheckFailure = retrieveFileContentChecks()
+        .fold(initial = false) { isFailing, check ->
+          val checkFailed = checkProhibitedContent(
+            repoPath = repoPath,
+            searchFiles = searchFiles,
+            prohibitedContentObj = check,
+          )
+          return@fold isFailing || checkFailed
```

Comment:
Use Kotlin idiomatic way

1. explicitly return - in this, we use a qualifier with the `return` like we are doing right now `return@fold`. 
2. implicit return - 
```suggestion
          isFailing || checkFailed
```

return statement without qualifier is  - return from the innermost fun (ignoring lambdas).

Reply:
This looks really neat. Thanks.
Now updated, PTAL.